### PR TITLE
Worst case redeem benchmark

### PR DESCRIPTION
**Pull request by daniel-savu**
Uses lend tokens as collateral in the redeem benchmarks, and re-generates the weights. This PR depends on https://github.com/interlay/interbtc/pull/1052.

### CODE REVIEW

File: crates/redeem/src/benchmarking.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#![cfg(feature = "runtime-benchmarks")]
```
Comment (gregdhill) [inline crates/redeem/src/benchmarking.rs:1]: Not required since it is already specified in `lib.rs`

File: crates/redeem/src/mock.rs [hunk 2]
```diff
@@ -343,6 +351,15 @@ impl ExtBuilder {
         .assimilate_storage(&mut storage)
         .unwrap();
 
+        GenesisBuild::<Test>::assimilate_storage(
+            &loans::GenesisConfig {
+                max_exchange_rate: Rate::from_inner(DEFAULT_MAX_EXCHANGE_RATE),
+                min_exchange_rate: Rate::from_inner(DEFAULT_MIN_EXCHANGE_RATE),
+            },
```
Comment (gregdhill) [inline crates/redeem/src/mock.rs:358]: Wondering if we can implement `Default` for this at source to use some sensible values then we don't need to re-declare this everywhere

File: crates/redeem/src/benchmarking.rs [hunk 3]
```diff
@@ -174,6 +180,7 @@ pub mod benchmarks {
         mint_wrapped::<T>(&caller, amount.into());
 
         mint_collateral::<T>(&vault_id.account_id, 100_000u32.into());
+        mint_vault_collateral::<T>(&vault_id);
```
Comment (gregdhill) [inline crates/redeem/src/benchmarking.rs:183]: Does this not replace the call in the preceding line?

File: crates/redeem/src/benchmarking.rs [hunk 4]
```diff
@@ -111,16 +116,17 @@ fn test_request<T: crate::Config>(vault_id: &DefaultVaultId<T>) -> DefaultRedeem
...
-    VaultId::new(
-        account("Vault", 0, 0),
-        get_collateral_currency_id::<T>(),
-        get_wrapped_currency_id::<T>(),
-    )
+fn mint_vault_collateral<T: crate::Config + loans::Config>(vault_id: &DefaultVaultId<T>) {
+    if vault_id.collateral_currency().is_lend_token() {
+        mint_lend_tokens::<T>(&vault_id.account_id, vault_id.collateral_currency());
+    } else {
+        mint_collateral::<T>(&vault_id.account_id, 100_000u32.into());
```
Comment (gregdhill) [inline crates/redeem/src/benchmarking.rs:123]: Actually, this is not required right?

Comment (daniel-savu) [reply crates/redeem/src/benchmarking.rs:123]: It's not required right now, I just wanted to make it easy to change the collateral used by the vault in the future

**Pull request merged**